lib/graph_gateway.go

```diff
@@ -276,36 +251,11 @@ Implementation iteration plan:
 */
 
 func (api *GraphGateway) loadRequestIntoSharedBlockstoreAndBlocksGateway(ctx context.Context, path string) (gateway.IPFSBackend, func(), error) {
-	bstore := api.bstore
-	carFetchingExch := newInboundBlockExchange()
-	doneWithFetcher := make(chan struct{}, 1)
-	exch := &handoffExchange{
-		startingExchange: carFetchingExch,
-		followupExchange: &blockFetcherExchWrapper{api.blockFetcher},
-		bstore:           bstore,
-		handoffCh:        doneWithFetcher,
-		metrics:          api.metrics,
-	}
-
-	notifierKey := api.getRootOfPath(path)
-	var notifier *notifiersForRootCid
-	for {
-		notifiers, _ := api.notifiers.LoadOrStore(notifierKey, &notifiersForRootCid{notifiers: []Notifier{}})
-		if n, ok := notifiers.(*notifiersForRootCid); ok {
-			n.lk.Lock()
-			// could have been deleted after our load. try again.
-			if n.deleted != 0 {
-				n.lk.Unlock()
-				continue
-			}
-			notifier = n
-			n.notifiers = append(n.notifiers, exch)
-			n.lk.Unlock()
-			break
-		} else {
-			return nil, nil, errors.New("failed to get notifier")
-		}
+	bstore, err := NewCacheBlockStore(1024)
```

Comment:
:warning: @willscott @aarshkshah1992 @aschmahmann

Just a small ask that **if we make breaking change like this, we should also break (rename) metrics**, otherwise we end up with confusing history making it impossible to reason about the system.

To illustrate, the meaning of `blockstore_cache_hit` and `blockstore_cache_requests` changed, it is no longer "global hit/miss", but "hit per single http request".

So the metric is broken, BUT on existing  Grafana it looks like we've only lost about ~10% of cache hits:

>  ![2023-06-22_19-43](https://github.com/ipfs/bifrost-gateway/assets/157609/95d2bf2e-2e8b-4be4-8fa1-a69e3c16c547)

iiuc this is nonsense, because we are now measuring totally different thing, right?